Run MAUI app on M1 Mac without Rosetta 2

### Description

I am trying to run a MAUI app on my M1 MacBook Pro. I don't have Rosetta 2 installed and when running `dotnet new maui && dotnet build -t:Run -f net6.0-maccatalyst` I get the following error:

`An error occurred trying to start process '/usr/local/share/dotnet/packs/Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-x64/6.0.1/Sdk/../tools/mono-aot-cross' with working directory '/Users/collin/Programmieren/dotNET/babi-desktop'. Bad CPU type in executable`

It seems like MAUI doesn't support the ARM architecture. Is this true?

### Steps to Reproduce

1. Create a new MAUI project on a Mac with an M1 processor (e.g. with `dotnet new maui`)
2. Run it (`dotnet build -t:Run -f net6.0-maccatalyst`)

### Version with bug

Preview 12 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

macOS, I was *not* able test on other platforms

### Affected platform versions

macOS 12.1

### Did you find any workaround?

Install Rosetta 2. 

### Relevant log output

```shell
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018: The "AOTCompile" task failed unexpectedly. [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018: System.AggregateException: One or more errors occurred. (One or more errors occurred. (An error occurred trying to start process '/usr/local/share/dotnet/packs/Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-x64/6.0.1/Sdk/../tools/mono-aot-cross' with working directory '/Users/collin/Programmieren/dotNET/babi-desktop'. Bad CPU type in executable)) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:  ---> System.AggregateException: One or more errors occurred. (An error occurred trying to start process '/usr/local/share/dotnet/packs/Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-x64/6.0.1/Sdk/../tools/mono-aot-cross' with working directory '/Users/collin/Programmieren/dotNET/babi-desktop'. Bad CPU type in executable) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:  ---> System.ComponentModel.Win32Exception (86): An error occurred trying to start process '/usr/local/share/dotnet/packs/Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-x64/6.0.1/Sdk/../tools/mono-aot-cross' with working directory '/Users/collin/Programmieren/dotNET/babi-desktop'. Bad CPU type in executable [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Diagnostics.Process.Start() [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at Xamarin.Utils.Execution.<>c__DisplayClass24_1.<RunAsync>b__0() in /Users/builder/azdo/_work/1/s/xamarin-macios/tools/common/Execution.cs:line 103 [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018: --- End of stack trace from previous location --- [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at Xamarin.MacDev.Tasks.XamarinTask.ExecuteAsync(String fileName, IList`1 arguments, String sdkDevPath, Dictionary`2 environment, Boolean mergeOutput, Boolean showErrorIfFailure, String workingDirectory) in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/XamarinTask.cs:line 96 [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    --- End of inner exception stack trace --- [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.Tasks.Task`1.get_Result() [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at Xamarin.MacDev.Tasks.AOTCompileTaskBase.<>c__DisplayClass32_0.<Execute>b__6(Task`1 v) in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AOTCompileTaskBase.cs:line 102 [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke() [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018: --- End of stack trace from previous location --- [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    --- End of inner exception stack trace --- [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.Tasks.Task.WaitAllCore(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at System.Threading.Tasks.Task.WaitAll(Task[] tasks) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at Xamarin.MacDev.Tasks.AOTCompileTaskBase.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AOTCompileTaskBase.cs:line 109 [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at Xamarin.MacDev.Tasks.AOTCompile.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks/Tasks/AOTCompile.cs:line 16 [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.0.101-preview.11.551/targets/Xamarin.Shared.Sdk.targets(898,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/Users/collin/Programmieren/dotNET/babi-desktop/babi-desktop.csproj]
```

### Depends on
- [x] https://github.com/xamarin/xamarin-macios/issues/15492
- [ ] https://github.com/dotnet/runtime/issues/82495
- [x] https://github.com/dotnet/runtime/pull/89961

--- a/src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.json.in
+++ b/src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.json.in
@@ -207,8 +207,9 @@
         "win-x64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-x86",
         "win-arm64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-x86",
         "linux-x64": "Microsoft.NETCore.App.Runtime.AOT.linux-x64.Cross.android-x86",
+        "linux-arm64": "Microsoft.NETCore.App.Runtime.AOT.linux-arm64.Cross.android-x86",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-x86",
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-x86"
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.android-x86"
       }
     },
     "Microsoft.NETCore.App.Runtime.AOT.Cross.android-x64": {
@@ -218,8 +219,9 @@
         "win-x64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-x64",
         "win-arm64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-x64",
         "linux-x64": "Microsoft.NETCore.App.Runtime.AOT.linux-x64.Cross.android-x64",
+        "linux-arm64": "Microsoft.NETCore.App.Runtime.AOT.linux-arm64.Cross.android-x64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-x64",
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-x64"
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.android-x64"
       }
     },
     "Microsoft.NETCore.App.Runtime.AOT.Cross.android-arm": {
@@ -229,8 +231,9 @@
         "win-x64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-arm",
         "win-arm64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-arm",
         "linux-x64": "Microsoft.NETCore.App.Runtime.AOT.linux-x64.Cross.android-arm",
+        "linux-arm64": "Microsoft.NETCore.App.Runtime.AOT.linux-arm64.Cross.android-arm",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-arm",
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-arm"
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.android-arm"
       }
     },
     "Microsoft.NETCore.App.Runtime.AOT.Cross.android-arm64": {
@@ -240,8 +243,9 @@
         "win-x64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-arm64",
         "win-arm64": "Microsoft.NETCore.App.Runtime.AOT.win-x64.Cross.android-arm64",
         "linux-x64": "Microsoft.NETCore.App.Runtime.AOT.linux-x64.Cross.android-arm64",
+        "linux-arm64": "Microsoft.NETCore.App.Runtime.AOT.linux-arm64.Cross.android-arm64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-arm64",
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.android-arm64"
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.android-arm64"
       }
     },
     "Microsoft.NETCore.App.Runtime.Mono.maccatalyst-arm64": {
@@ -285,7 +289,7 @@
       "version": "${PackageVersion}",
       "alias-to": {
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.tvos-arm64",
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.tvos-arm64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.tvos-arm64",
       }
     },
     "Microsoft.NETCore.App.Runtime.Mono.tvos-arm64" : {
@@ -304,7 +308,7 @@
       "kind": "Sdk",
       "version": "${PackageVersion}",
       "alias-to": {
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-arm64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.maccatalyst-arm64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-arm64"
       }
     },
@@ -312,7 +316,7 @@
       "kind": "Sdk",
       "version": "${PackageVersion}",
       "alias-to": {
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-x64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.maccatalyst-x64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.maccatalyst-x64"
       }
     },
@@ -320,7 +324,7 @@
       "kind": "Sdk",
       "version": "${PackageVersion}",
       "alias-to": {
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.tvossimulator-arm64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.tvossimulator-arm64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.tvossimulator-arm64"
       }
     },
@@ -328,7 +332,7 @@
       "kind": "Sdk",
       "version": "${PackageVersion}",
       "alias-to": {
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.tvossimulator-x64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.tvossimulator-x64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.tvossimulator-x64"
       }
     },
@@ -337,14 +341,14 @@
       "version": "${PackageVersion}",
       "alias-to": {
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.ios-arm64",
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.ios-arm64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.ios-arm64",
       }
     },
     "Microsoft.NETCore.App.Runtime.AOT.Cross.iossimulator-arm64": {
       "kind": "Sdk",
       "version": "${PackageVersion}",
       "alias-to": {
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.iossimulator-arm64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.iossimulator-arm64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.iossimulator-arm64"
       }
     },
@@ -352,7 +356,7 @@
       "kind": "Sdk",
       "version": "${PackageVersion}",
       "alias-to": {
-        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.iossimulator-x64",
+        "osx-arm64": "Microsoft.NETCore.App.Runtime.AOT.osx-arm64.Cross.iossimulator-x64",
         "osx-x64": "Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.iossimulator-x64"
       }
     },

